src/legacy/ui/public/index_patterns/index_patterns_api_client.js

```diff
@@ -17,96 +17,85 @@
  * under the License.
  */
 
-import { resolve as resolveUrl, format as formatUrl } from 'url';
+import { kfetch } from '../kfetch';
 
-import { pick, mapValues } from 'lodash';
+import { IndexPatternMissingIndices } from './errors';
 
-import { IndexPatternMissingIndices } from '../errors';
+function join(...uriComponents) {
+  return uriComponents.filter(Boolean).map(encodeURIComponent).join('/');
+}
 
-export function createIndexPatternsApiClient($http, basePath) {
-  const apiBaseUrl = `${basePath}/api/index_patterns/`;
+function request(method, url, query, body) {
+  return kfetch({
+    method,
+    pathname: url,
+    query,
+    body,
+  })
+    .catch((resp) => {
+      if (resp.body.statusCode === 404 && resp.body.statuscode === 'no_matching_indices') {
+        throw new IndexPatternMissingIndices(resp.body.message);
+      }
 
-  function join(...uriComponents) {
-    return uriComponents.filter(Boolean).map(encodeURIComponent).join('/');
-  }
+      const err = new Error(resp.body.message || resp.body.error || `${resp.body.statusCode} Response`);
+      err.status = resp.body.statusCode;
+      err.body = resp.body.message;
+      throw err;
+    });
+}
 
-  function getUrl(path, query) {
-    const noNullsQuery = pick(query, value => value != null);
-    const noArraysQuery = mapValues(noNullsQuery, value => (
-      Array.isArray(value) ? JSON.stringify(value) : value
-    ));
+export class IndexPatternsApiClient {
+  constructor(basePath) {
+    this.apiBaseUrl = `${basePath}/api/index_patterns/`;
+  }
 
-    return resolveUrl(apiBaseUrl, formatUrl({
-      pathname: join(...path),
-      query: noArraysQuery,
-    }));
+  _getUrl(path) {
+    return this.apiBaseUrl + join(path);
```

Comment:
This should be `join(...path)`. This has broken creating roll-up index patterns.